### PR TITLE
🌱 Use `go install` to have self contained binaries

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://github.com/syself/open-source/issues/35

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,58 +92,61 @@ export KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT ?= 60s
...
-	cd $(TOOLS_DIR) && go build -mod=vendor -tags=tools -o $(CTLPTL) github.com/tilt-dev/ctlptl/cmd/ctlptl
+$(CTLPTL):
+	go install github.com/tilt-dev/ctlptl/cmd/ctlptl@v0.8.20
 
 CLUSTERCTL := $(abspath $(TOOLS_BIN_DIR)/clusterctl)
 clusterctl: $(CLUSTERCTL) ## Build a local copy of clusterctl
-$(CLUSTERCTL): $(TOOLS_DIR)/go.mod 
-	cd $(TOOLS_DIR) && go build -mod=vendor -tags=tools -o $(CLUSTERCTL) sigs.k8s.io/cluster-api/cmd/clusterctl
+$(CLUSTERCTL): $(TOOLS_DIR)/go.mod
+	go install sigs.k8s.io/cluster-api/cmd/clusterctl@v1.4.3
```
Comment (batistein) [inline Makefile:126]: @aniruddha2000 is here the versioning problem fixed when running the cli tool?

Comment (guettli) [reply Makefile:126]: @aniruddha2000 please use this like in HV:

```
$(CLUSTERCTL):
	curl -sSLf https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.3/clusterctl-linux-amd64 -o $(CLUSTERCTL)
	chmod a+rx $(CLUSTERCTL)
```

**Review by guettli — CHANGES_REQUESTED** [review]
Please clean tools.go, and remove lines which are not needed any more. Thank you.

**Review by guettli — APPROVED** [review]
lgtm

**Pull request merged**